pkg/hostagent/requirements.go

```diff
@@ -170,6 +170,19 @@ Also see "/var/log/cloud-init-output.log" in the guest.
 `,
 			})
 	}
+	if *a.y.SSH.ForwardAgent {
+		req = append(req, requirement{
+			description: "linking ssh auth socket to /tmp/ssh-auth/agent.sock",
+			script: `#!/bin/bash
+set -eux -o pipefail
+if ! timeout 30s bash -c "mkdir -p -m 700 /tmp/ssh-auth && ln -f $SSH_AUTH_SOCK /tmp/ssh-auth/agent.sock"; then
+	echo >&2 "unable to link $SSH_AUTH_SOCK to /tmp/ssh-auth/agent.sock"
+	exit 1
+fi
+`,
+			debugHint: `Linking $SSH_AUTH_SOCK to /tmp/ssh-auth/agent.sock failed.`,
+		})
+	}
```

Comment:
I think this should be in `$XDG_RUNTIME_DIR/lima`

Reply:
The whole point of this PR is to have a predictable address, that you can use in scripts, so you don't have to run `$(lima printenv SSH_AUTH_SOCK)` to get the filename.

Putting the link into another location that can only be determined by shelling into the VM makes the whole PR pointless IMO.

Reply:
However, I just noticed that this PR should include documentation (assuming we can agree on a fixed location).

Just for reference, ssh also puts the agent socket into `/tmp`:

```console
$ lima printenv SSH_AUTH_SOCK
/tmp/ssh-XXXXv2u4Qd/agent.2043
```

So linking it to `/tmp/ssh-auth/agent.sock` doesn't feel wrong to me.

Reply:
$XDG_RUNTIME_DIR/lima (/run/user/$UID/lima) is still predictable when you can get the UID

Reply:
Generally, creating a dir under /tmp with a predictable name is known as a TOCTOU attack vector for multi-user system.

Lima isn't really expected to be used for multi-user VM, but I'm little a bit on the fence to adopt that vulnerable-ish pattern.

Reply:
Using /run/lima is ok for me though

Reply:
`XDG_RUNTIME_DIR` being the same as `/run/user/$UID` seems based on common usage and does not seem to be guaranteed by the spec.

`/run/lima` is likely on a different filesystem, so we would have to use a symlink instead of a hard link (probably not a problem though).

However, I don't see any TOCTOU attack that doesn't already apply to the original socket location. If `/tmp/ssh-XXXXv2u4Qd/agent.2043` is secure, then `/tmp/ssh-auth/agent.sock` should be as well.

As for the `/run/lima` name, I would prefer a name that doesn't include `lima` because I may want to use the same location in the future with WSL2, where using `/run/lima` would be confusing.

Reply:
So theoretically I see a vulnerability because we don't check the ownership of the directory before creating the link in there. The same vulnerability would however apply to any other file system location since we are using a predictable name.

But practically speaking, given that we allow password-less `sudo`, none of this is preventing a targeted attack.

We are just trying to avoid accidental exposure to workloads users are running inside the VM, ideally in containers.

Reply:
Avoiding TOCTOU:

```console
jan@lima-default:/tmp$ ln -s -f $(dirname $SSH_AUTH_SOCK) /tmp/ssh-auth
jan@lima-default:/tmp$ ln -s -f $SSH_AUTH_SOCK /tmp/ssh-auth/agent.sock
jan@lima-default:/tmp$ ls -l /tmp/ssh-auth/agent.sock
lrwxrwxrwx 1 jan jan 30 Jan 29 06:10 /tmp/ssh-auth/agent.sock -> /tmp/ssh-XXXXv2u4Qd/agent.2043
```

But in this case having the additional directory level is pointless, so we could go back to:

```console
jan@lima-default:/tmp$ ln -s -f $SSH_AUTH_SOCK /tmp/SSH_AUTH_SOCK
jan@lima-default:/tmp$ ls -l /tmp/SSH_AUTH_SOCK
lrwxrwxrwx 1 jan jan 30 Jan 29 06:12 /tmp/SSH_AUTH_SOCK -> /tmp/ssh-XXXXv2u4Qd/agent.2043
```

Reply:
FWIW, using a symlink was my original suggestion in https://github.com/rancher-sandbox/rancher-desktop/issues/3488#issuecomment-1406936969. It avoids all the permissions problems because the original permissions of the target still apply.